pkg/bgpv1/gobgp/diffstore.go

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Yes. I did consider this. Even had a PR out for it #22212. There are two reasons for having it in the gobgp namespace. 

First is that we use this custom `Signaler` instead of a generic mechanism, so it is just easier for now.

Second is that this DiffStore shouldn't be the long term solution, not something we want to promote if not necessary. It is a temporary solution to do differential updates within the current architecture, but long term we want to move to using `Resource[T]` directly, which will improve memory overhead and add retry logic.

Reply:
Gotchya.